packages/common/container-definitions/src/deltas.ts

```diff
@@ -144,7 +145,7 @@ export interface IDeltaManagerEvents extends IEvent {
 	 *
 	 * - `reason`: Describes the reason for which the delta manager was disconnected.
 	 */
-	(event: "disconnect", listener: (reason: string) => void);
+	(event: "disconnect", listener: (reason: string, props?: ITelemetryProperties) => void);
```

Comment:
I think the ideal place to be - replace `reason` string with error object. 
That's clearly hard here as we over-expose IDeltaManager to runtime (including this event).
So it's easier to add new argument instead of changing structure.
But everywhere else I'd expect to simply do such replacement, and have just one argument

Reply:
This assumes that we could wrap some strings into errors, i.e. "Switch to write". I think it's fine to convert it to error object (in order to follow what I've written above).
That said, it has some negatives (as technically it's not an error). Curious what you think about it?

Reply:
Yes, I was not in favor of replacing reason string with error as "disconnect" will not always be due to errors and we don't want to give that illusion to the user. And as you pointed out in that case, we will need to specifically replace string with Errors which seems unnecessary, and it would also be a breaking change. So really don't see positives in doing that.

Reply:
However, I can replace props with error object and then while logging extract props from that error instead before.

Reply:
But props does gives us some more flexibility over the error as we can add things in the future? @markfields

Reply:
There will definitely be overlap between the reason and the error.  I think my lean would be to keep them separate though.  Some reasons will come with errors, some will not.

As mentioned below, I think we should pass the error itself around, not the `props` form of it.  This way we'll get the `Data_error` field as we're accustomed to.

Reply:
Changed it to error.